src/api/broker/accounts.rs

```diff
@@ -1,88 +1,140 @@
-        #[serde(skip_serializing_if = "Option::is_none")]
-        pub contact: Option<Contact>,
-        #[serde(skip_serializing_if = "Option::is_none")]
-        pub identity: Option<Identity>,
-        #[serde(skip_serializing_if = "Option::is_none")]
-        pub disclosures: Option<Disclosures>,
-        #[serde(skip_serializing_if = "Option::is_none")]
-        pub trusted_contact: Option<TrustedContact>
+    pub async fn data(&mut self) -> Result<Account> {
+        if let Some(account) = self.data.as_ref().cloned() {
+            Ok(account)
+        } else {
+            let account = self.execute(GetAccount).await?;
+
+            self.data = Some(account.clone());
+
+            Ok(account)
+        }
     }
-}
 
-impl BrokerClient {
-    pub fn create_account(&self, contact: Contact, identity: Identity) -> CreateAccountBuilder {
-        CreateAccountBuilder(
-            self,
-            CreateAccount {
-                contact,
-                identity,
-                disclosures: Default::default(),
-                agreements: vec![],
-                documents: vec![],
-                trusted_contact: Default::default(),
-                enabled_assets: vec![],
-            },
-        )
+    pub fn get_data(&self) -> Option<&Account> {
+        self.data.as_ref()
     }
 
-    pub fn get_all_accounts(&self) -> GetAllAccountsBuilder {
-        GetAllAccountsBuilder(self, GetAllAccounts::default())
+    pub async fn execute<T: ClientEndpoint<Context = Self, Error = Error>>(
+        &self,
+        endpoint: T,
+    ) -> Result<T::Output> {
+        endpoint.run(self).await
     }
 }
 
-impl AccountView<'_> {
-    pub fn update(&self) -> UpdateAccountBuilder {
-        UpdateAccountBuilder(self, UpdateAccount::default())
+impl HttpClientContext for AccountView {
+    type Error = Error;
+
+    fn new_request(&self, method: Method, url: &str) -> Request {
+        self.client.new_request(method, url)
+    }
+
+    async fn run_request(&self, request: Request) -> Result<Response, Self::Error> {
+        self.client.run_request(request).await
     }
 }
 
-endpoint! {
-    impl GET "/accounts" = GetAllAccounts => Vec<SmallAccount> { |this, request| request.query(this).query(&[("query", this.query.join(" "))]) };
-    impl POST "/accounts" = CreateAccount => Account { |this, request| request.json(this) };
-    impl PATCH "/accounts" = UpdateAccount => Account { |this, request| request.json(this) }
-    | account (|_, account_id| format!("/accounts/{account_id}"));
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Default, ClientEndpoint)]
+#[endpoint(Get(empty) (format!("/accounts/{}", client.id())) in AccountView -> Account)]
+pub struct GetAccount;
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default, ClientEndpoint)]
+#[endpoint(Get(query) "/accounts" in BrokerClient -> Vec<SmallAccount>)]
+pub struct GetAllAccounts {
+    pub query: Vec<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
I guess now all occurrences of `#[serde(skip_serializing_if = "Option::is_none")]` can be replaced with `#[skip_serializing_none]` at the top of the struct.

Reply:
Not all, on some fields an Option<something> is something | null, that's just select cases, will fix that

Reply:
Done.